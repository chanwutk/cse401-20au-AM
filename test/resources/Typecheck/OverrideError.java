class OverrideError {
  public static void main(String[] args) {
    System.out.println(1);
  }
}

class Foo {
  public Foo m1(int n1, Foo n2) {
    return new Foo();
  }
}

class Foo2 extends Foo {
  // incompatible types
  public int m1(int n3, Foo2 n4) {
    return 2;
  }
}


class Foo3 {
  public Foo m1(int n1, Foo n2) {
    return new Foo();
  }
}

class Foo4 extends Foo3 {
  // wrong number of parameters
  public Foo m1(int n3, Foo n4, int a) {
    return new Foo();
  }
}

class Foo5 {
  public Foo m1(int n1, Foo n2) {
    return new Foo();
  }
}

class Foo6 extends Foo5 {
}

class Foo7 extends Foo6 {
  // wrong number of parameters from the base of base
  public Foo m1(int n3, Foo n4, int a) {
    return new Foo();
  }
}
