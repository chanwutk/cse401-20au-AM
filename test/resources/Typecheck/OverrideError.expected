OverrideError.java:15: error: incompatible types: int cannot be converted to Foo
OverrideError.java:15: error: incompatible types: Foo cannot be converted to Foo2
OverrideError.java:29: error: cannot override method with different number of parameters: expected 2 parameters, but received 3 parameters
OverrideError.java:45: error: cannot override method with different number of parameters: expected 2 parameters, but received 3 parameters
